### PR TITLE
chore: write explanatory comments on test functions in the sample contract

### LINKED ISSUE #317: Clarify the reason for changing the storage of the Number Contract using the query function, which is readOnly permission
The purpose of calling the function that changes storage from the query function is to detect errors in the integration test.
The relevant lines are:
https://github.com/Finschia/cosmwasm/blob/dynamic_link/contracts/call-number/src/contract.rs#L224-L226

### PR DESCRIPTION
**Pull request by 170210**
# Description
This PR has following changes.

- write explanatory comments on test functions in the sample contract

Closes #317 

## Types of changes

- [ ] Bug fix (changes which fixes an issue)
- [ ] New feature (changes which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] ETC (build, ci, docs, perf, refactor, style, test)

## Checklist

- [x] I followed the [contributing guidelines](https://github.com/Finschia/cosmwasm/blob/main/CONTRIBUTING.md).
- [x] I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.
- [x] The PR title and commits are followed [conventional commit form](https://www.conventionalcommits.org/en/v1.0.0).

### CODE REVIEW

**Review by loloicci — DISMISSED** [review]

**Comment by loloicci** [comment]
> change the proof.proto path to third_party/proto/confio

Please fix the description of this PR. (maybe mistake)

**Review by kokeshiM0chi — DISMISSED** [review]
Change the title prefix. Sorry for the late review. Is `comment:` better?

**Comment by zemyblue** [comment]
Please use `chore: ` structural elements.

**Review by da1suk8 — COMMENTED** [review]
It's a bit detailed, but I feel that "panic" makes it seem as if a panic is occurring, so it would be helpful if you could correct it to say that an error is occurring.

**Review by da1suk8 — APPROVED** [review]

**Review by loloicci — APPROVED** [review]

File: contracts/call-number/src/contract.rs [hunk 1]
```diff
@@ -241,6 +241,9 @@ fn query_number_dyn(deps: Deps) -> Result<NumberResponse, ContractError> {
     Ok(NumberResponse { value })
 }
 
+// This function is used to check if an appropriate error has been occurred in the VM,
+// when a caller with read-only permissions attempts to call a callable point with read/write permissions.
```
Comment (loloicci) [inline contracts/call-number/src/contract.rs:245]:
```suggestion
// This function is used to check if an appropriate error has been occurred in the VM
// when a caller with read-only permissions attempts to call a callable point with read/write permissions.
```

File: contracts/call-number/src/contract.rs [hunk 2]
```diff
@@ -254,6 +257,9 @@ fn query_add(deps: Deps, by: i32) -> Result<NumberResponse, ContractError> {
...
```
Comment (loloicci) [inline contracts/call-number/src/contract.rs:261]:
```suggestion
// This function is used to check if an appropriate error has been occurred in the VM
// when a caller with read-only permissions attempts to call a callable point with read/write permissions.
```

File: contracts/call-number/src/contract.rs [hunk 3]
```diff
@@ -267,6 +273,9 @@ fn query_sub(deps: Deps, by: i32) -> Result<NumberResponse, ContractError> {
...
```
Comment (loloicci) [inline contracts/call-number/src/contract.rs:277]:
```suggestion
// This function is used to check if an appropriate error has been occurred in the VM
// when a caller with read-only permissions attempts to call a callable point with read/write permissions.
```

**Review by loloicci — CHANGES_REQUESTED** [review]

**Review by da1suk8 — APPROVED** [review]

**Review by loloicci — APPROVED** [review]

**Pull request merged**